### ISSUE TITLE
Invalid char in username for some sites

### ISSUE THREAD

**Issue by enodr** [WebBreacher/WhatsMyName#673]
I have checked issue #55 and issue #430 but it looks like the problem of handling invalid chars like "." in usernames is done only for subdomains but not in the url.

One example:
https://wanelo.co/john is valid  but https://wanelo.co/john.doe is in fact the john account, thus john.doe is a false positive. The json data file should have an allowed/forbidden char list (both could be usefull). I'd be glad to contribute and add the changes to [wmn-data.json](https://github.com/WebBreacher/WhatsMyName/blob/main/wmn-data.json) if this trivial change is approved.

**Comment by enodr** [comment]
Yes I was thinking about a per site / entry option. Options to discuss:
- a good variable (key) name
- regex or just plain chars

For now the false positives I have had it would be enough to deal with a simple option like:
`badchars: '.'`
on a per site basis.

Example:

```json
       {
        "name" : "Wanelo",
        "uri_check" : "https://wanelo.co/{account}",
        "badchars" : ".",
        "post_body" : "",
        "e_code" : 200,
        "e_string" : "on Wanelo</title>",
        "m_string" : "Hmm, that's embarrassing",
        "m_code" : 404,
        "known" : ["lisandrareyes"],
        "cat" : "shopping",
        "valid" : true
       },
```

and code logic would be as simple as (python):

```python
badchars = set(site["badchars"])
if any((c in account) for c in badchars): continue
```

**Comment by WebBreacher** [comment]
So, the `.` is really the only character that I've noticed causes us problems since usernames can be in the subdomains or as a parameter. I'm wondering if the parameter could just be a Boolean `strip_bad_char` with values of `True` or `False`. If true, then remove anything non-[a-zA-Z0-9].

Thoughts?

**Comment by enodr** [comment]
Seems totally good!

**Comment by downerboy** [comment]
from my experience, adulterating the data in the long, run results and bad data sheet all data as if it's wholly yet poisonous or infectious i.e. Don't modify anything make a copy of it or something in memory, and work with it there sanitize it if you have to, but first make sure you're calling the correct function for handling the characters e.g. like the char function for C++ on windows, I believe it's W_char, and using.net C++ on window systems.  I know this is python and not C++ but what is python doing in the background Wireshark I know this is python and not C++ but what is python doing in the background, Wireshark will provide tremendous insight.  

what about the trading an extension method for reading the header of the source of the data to see if whoever is maintaining that site is aware enough to keep it to UTF-8 and if not see you at the code page they're saying it is run from the data through a test to verify if that is the actual code page it's coming across and not being ignored and it's actually UTF-8, which is the case most of the time, but there are times it's not. And I think creating and handling for each source uniquely is not realistic for maintenance and becomes so verbose that it will just become a giant spaghetti Mesa people just keep piling upon them piling upon because everyone's too scared to break it.

**Comment by WebBreacher** [comment]
OK @enodr. I'll take this on to insert the `strip_bad_char` boolean

**Comment by yooper** [comment]
@WebBreacher , we can make the field optional, such that if it is specified/provided the `strip_bad_char` method is called. That will save you the time of editing each entry in the `wmn-data.json` file.

**Comment by WebBreacher** [comment]
Agreed @yooper and, at the same time, we can do the same for the `"post_body" : "",` which is rarely used.

**Comment by yooper** [comment]
I will put in sometime this week and make it happen.

**Comment by WebBreacher** [comment]
Hey thanks! I can mod the JSON if you wanna add the feature to the script.

**Comment by WebBreacher** [comment]
Going to close this as:
1. We have the `strip_bad_char` schema addition
2. The python code @yooper was going to mod has been removed from this project into its own project.
